packages/mdc-slider/foundation.ts

```diff
@@ -21,113 +21,123 @@
-      deregisterThumbContainerInteractionHandler: (/* type: string, handler: EventListener */) => {},
-      registerBodyInteractionHandler: (/* type: string, handler: EventListener */) => {},
-      deregisterBodyInteractionHandler: (/* type: string, handler: EventListener */) => {},
-      registerResizeHandler: (/* handler: EventListener */) => {},
-      deregisterResizeHandler: (/* handler: EventListener */) => {},
-      notifyInput: () => {},
-      notifyChange: () => {},
-      setThumbContainerStyleProperty: (/* propertyName: string, value: string */) => {},
-      setTrackStyleProperty: (/* propertyName: string, value: string */) => {},
-      setMarkerValue: (/* value: number */) => {},
-      appendTrackMarkers: (/* numMarkers: number */) => {},
-      removeTrackMarkers: () => {},
-      setLastTrackMarkersStyleProperty: (/* propertyName: string, value: string */) => {},
-      isRTL: () => /* boolean */ false,
-    });
+  static get defaultAdapter(): MDCSliderAdapter {
+    // tslint:disable:object-literal-sort-keys
+    return {
+      hasClass: () => false,
+      addClass: () => undefined,
+      removeClass: () => undefined,
+      getAttribute: () => null,
+      setAttribute: () => undefined,
+      removeAttribute: () => undefined,
+      computeBoundingRect: () => ({top: 0, right: 0, bottom: 0, left: 0, width: 0, height: 0}),
+      getTabIndex: () => 0,
+      registerInteractionHandler: () => undefined,
+      deregisterInteractionHandler: () => undefined,
+      registerThumbContainerInteractionHandler: () => undefined,
+      deregisterThumbContainerInteractionHandler: () => undefined,
+      registerBodyInteractionHandler: () => undefined,
+      deregisterBodyInteractionHandler: () => undefined,
+      registerResizeHandler: () => undefined,
+      deregisterResizeHandler: () => undefined,
+      notifyInput: () => undefined,
+      notifyChange: () => undefined,
+      setThumbContainerStyleProperty: () => undefined,
+      setTrackStyleProperty: () => undefined,
+      setMarkerValue: () => undefined,
+      appendTrackMarkers: () => undefined,
+      removeTrackMarkers: () => undefined,
+      setLastTrackMarkersStyleProperty: () => undefined,
+      isRTL: () => false,
+    };
+    // tslint:enable:object-literal-sort-keys
   }
 
+  private rect_: ClientRect | null = null;
+
+  /**
+   * We set this to NaN since we want it to be a number, but we can't use '0' or '-1'
+   * because those could be valid tabindices set by the client code.
+   */
+  private savedTabIndex_ = NaN;
+
+  private active_ = false;
+  private inTransit_ = false;
+  private isDiscrete_ = false;
+  private hasTrackMarker_ = false;
+  private handlingThumbTargetEvt_ = false;
+  private min_ = 0;
+  private max_ = 100;
+  private step_ = 0;
+  private value_ = 0;
+  private disabled_ = false;
+  private preventFocusState_ = false;
+
+  private readonly thumbContainerPointerHandler_: EventListener;
```

Comment:
this could also be SpecificEventListener, right?

Reply:
Done